Fixed the breaks by New version of ModelingToolkit (#139)
JuliaIntervals/IntervalConstraintProgramming.jl: docs/src/index.md

([-100, 100],[-100, 100])
```

> ### Without using Macros

Reviewer: macros

Reviewer 2: ok.

We can also make an object `S`, of type `Separator` or `C`, of type `Contractor` without using Macros, for that you need to define variables using `ModelingToolkit.jl`.
Example  

```julia
julia> using IntervalConstraintProgramming, IntervalArithmetic, ModelingToolkit

julia> @variables x y
(x(), y())

julia> S = Separator(x+y<1)
Separator:
  - variables: x, y
  - expression: x() + y() == [-∞, 1]

julia> C = Contractor(x+y)
 Contractor in 2 dimensions:
   - forward pass contracts to 1 dimensions
   - variables: Symbol[:x, :y]
   - expression: x() + y()
```

While making `Separator`or `Contractor`'s object we can also specify variables, like this

```julia
julia> vars = @variables x y z
(x(), y(), z())

julia> S = Separator(vars, x+y<1)
Separator:
  - variables: x, y, z
  - expression: x() + y() == [-∞, 1]

julia> C = Contractor(vars, y+z)
Contractor in 3 dimensions:
  - forward pass contracts to 1 dimensions
  - variables: Symbol[:x, :y, :z]
  - expression: y() + z()
```

> We can make object (of type `Separator` or `Contractor`)by just using function name (Note: you have to specify variables explicitly as discussed above when you make objects by using function name). We can also use polynomial function to make objects.

Reviewer 3: ```suggestion
We can make objects (of type `Separator` or `Contractor`) by just using the function name (Note: you have to specify variables explicitly as discussed above in this case). We can also use a polynomial function as shown below.
```

```julia
julia> vars=@variables x y
(x(), y())

julia> f(a,b)= a+b
f (generic function with 1 method)

julia> C = Contractor(vars,f)
Contractor in 2 dimensions:
  - forward pass contracts to 1 dimensions
  - variables: Symbol[:x, :y]
  - expression: x() + y()

julia> f(a,b) = a+b <1
f (generic function with 1 method)

julia> S=Separator(vars, f)
Separator:
  - variables: x, y
  - expression: x() + y() == [-∞, 1]  

julia> using DynamicPolynomials       #using polynomial functions

julia> pvars = @polyvar x y
(x, y)

julia> f(a,b) = a + b < 1
p (generic function with 1 method)

julia> S=Separator(pvars, f)
Separator:
  - variables: x, y
  - expression: x() + y() == [-∞, 1]
```
#### BasicContractor

> Object of type `Contractor` have four feilds (variables, forward, backward and expression), among them data of two feilds (forward, backward) are useful (i.e forward and backward functions) for further usage of that object, thats why it is preferred to use an object of type `BasicContractor` in place of `Contractor` which only contain these two feilds for less usage of memory by unloading all the extra stuff.(Note: Like object of `Contractor` type,`BasicContractor`'s object will also have all the properties which are discussed above).

Reviewer 3: ```suggestion
Objects of type `Contractor` have four fields (variables, forward, backward and expression), among them data of two fields (forward, backward) are useful (i.e. forward and backward functions) for further usage of that object, that's why it is preferred to use an object of type `BasicContractor` in place of `Contractor` which only contain these two fields for less usage of memory by unloading all the extra stuff. (Note: Like object of `Contractor` type,`BasicContractor`'s object will also have all the properties which are discussed above).
```

Reviewer 2: That's a very silly mistake, thanks for pointing it out.

Reviewer 3: > thats why it is preferred to use an object of type `BasicContractor` in place of `Contractor`

In which context is it preferred? Like, from the user perspective, for library code? I think that dwelling a bit into some of these details, or giving another example, would improve that paragraph.

Also i guess that the other two fields of `Contractor` do have a reason to exist and they are "useful" in some sense (at least internally). I think that this sentence could be improved a bit: "... only contain these two fields for less usage of memory by unloading all the extra stuff..".

Nice work!

Reviewer 2: As I already mentioned in the paragraph, using `BasicContractor` is better because of less usage of memory and for less time complexity.
Other two fields are their just to give information about variables and expression used, which has no usage in contracting the interval w.r.t given constraint for which the object is used which is done by just using forward and backward function. Yes, Internally information about variables and expression is necessary after all only they are used to make forward and backward function but finally we just need these two functions. And these forward and backward functions are present in object .of type `BasicContractor`.
